app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -112,25 +112,62 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
 <%= "          --data 'remote_execution_interface=#{@remote_execution_interface}' \\\n" if @remote_execution_interface.present? -%>
 <%= "          --data 'packages=#{@packages}' \\\n" if @packages.present? -%>
 
-}
+    }
 
     <% if @force -%>
     yum remove -y katello-ca-consumer*
     <% end -%>
 
-    CONSUMER_RPM=$(mktemp --suffix .rpm)
-    curl --silent --show-error --output $CONSUMER_RPM <%= subscription_manager_configuration_url(hostname: @url_host) %>
+    KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/katello-server-ca.crt
+    RHSM_CFG=/etc/rhsm/rhsm.conf
 
-    # Workaround for systems with enabled FIPS,
-    # where installation of RPM generated on RHEL7 cause 'no digest' error
-    # See https://projects.theforeman.org/issues/32068
-    if [ "$(cat /proc/sys/crypto/fips_enabled)" = "1" ]; then
-        rpm -ivh --nodigest --nofiledigest $CONSUMER_RPM
+    # Install or upgrade subscription-manager
+    if ! [ -x "$(command -v subscription-manager)" ] ; then
+      if [ -f /usr/bin/dnf ]; then
+        dnf install -y subscription-manager
+      else
+        yum install -y subscription-manager
+      fi
     else
-        yum localinstall $CONSUMER_RPM -y
+      if [ -f /usr/bin/dnf ]; then
+        dnf upgrade -y subscription-manager
+      else
+        yum upgrade -y subscription-manager
+      fi
+    fi
+
+    if ! [ -f $RHSM_CFG ] ; then
+      echo "'$RHSM_CFG' not found, cannot configure subscription-manager"
+      exit 1
```

Comment:
This `exit` means the cleanup below never triggers? Or am I reading the flow wrongly?

Reply:
I think you are reading this correctly.